kernel/api/src/main/java/org/sakaiproject/tasks/api/TaskService.java

```diff
@@ -23,13 +23,18 @@
 
     UserTask createSingleUserTask(UserTaskAdapterBean transfer);
     UserTask saveUserTask(UserTaskAdapterBean transfer);
+    UserTask saveUserTask(Task task, String userId, Integer priority, String notes);
```

Comment:
Why not use UserTaskAdapterBean here? It has the priority and notes fields contained in it.

Reply:
The existing method "saveUserTask" was doing more stuff than only saving the info of the database entity "USER_TASKS", since it was saving also the Task. I didn't want to do that, so I implemented a method that only worked with USER_TASKS and the relevant information within this database entity, and named it like it should. I didn't want to re-implement or rename existing code, so I took the decission to overload this method name.

Reply:
Why didn't you want to save the task? It's in the service layer, so multiple db operations are perfectly valid. UserTaskAdapterBean basically has all the information you need to save the task, so adding another method is just confusing.

Reply:
Also, don't keep resolving these before people have a chance to look at your comments. When we click the link in the email, we have to go and hunt the resolved block down. It's irritating.

Reply:
I don't want to save the task because what I need is ONLY to save users personal notes/priorities. That's the entity UserTask. Now, some tasks are created by instructors or admin and then they're delivered to students. Students don't have to update tasks, because it would affect to every student on a site or a group. They only can update their own data related to the task, and that's the UserTask entity.

Reply:
Ok, so why not just add a boolean flag to not update the Task? I hear what you're saying, but if you think this area of the code needs a rewrite, then let's do that rather than confusing things further.

Reply:
I wrote this code, but I can't remember why I save the Task there - I'd have to revisit the code myself. However, I think having this extra method is a code smell. Let's fix the existing piece instead of just rushing it.

Reply:
You can revisit your code and rename it to some other method name if you want, since it does not save "UserTasks". It does things beyond this simple action. My method does what its name indicates, using primitives in a logic and understandable way.

Reply:
It saves a usertask, what else does it do? Basically, you are being paid to do this work, you have seen an issue in the existing code and are introducing an overloaded method as a way to get around that, instead of fixing it. Why not do the work and fix it, improve what's there already, instead of hacking it?

Reply:
When I say you are being paid, I mean you have time and money to do it right, and I don't think this is doing it right.

Reply:
I think you'll be happy with my latest change... 💸 ⌚ 😄